manifests/grubby.pp

```diff
@@ -0,0 +1,21 @@
+# Private class.
+class kernel::grubby{
+  if $caller_module_name != $module_name {
+    fail("Use of private class ${name} by ${caller_module_name}")
+  }
+
+  $grub_default_kernel = $kernel::grub_default_kernel ? {
+    'UNSET' => $kernel::kernel_version,
+    default => $kernel::grub_default_kernel,
+  }
+
+  # Under what circumstances could $grub_default_kernel *still*  == 'UNSET'?
```

Comment:
If the `$kernel::version` parameter is `UNSET` (default) and `$kernel::grub_default_kernel` is `UNSET` (default) then the `$grub_default_kernel` scoped to this class would be `UNSET`.